kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/develop/WatchMojo.java

```diff
@@ -171,6 +176,7 @@ protected KitLogger createLogger(String prefix) {
                 .serviceHubFactory(serviceHubFactory)
                 .hub(hub)
                 .dispatcher(getLogDispatcher(hub))
+                .postGoalTask(s -> MavenUtil.callMavenPluginWithGoal(project, session, pluginManager, s))
```

Comment:
After https://github.com/eclipse/jkube/pull/402/files#r495933135

I think that you could remove the redundant configurations from the WatchContext.

The WatchContext should have the field for postGoalTask like
```java
// private String postGoal; REMOVED
private Supplier<Boolean> postGoalTask;
```
```suggestion
                .postGoalTask(() -> MavenUtil.callMavenPluginWithGoal(project, session, pluginManager, watchPostGoal))
```